base/client.jl

```diff
@@ -231,9 +231,18 @@ let reqarg = Set(UTF8String["--home",          "-H",
                             "--bind-to"])
     global process_options
     function process_options(opts::JLOptions, args::Vector{UTF8String})
-        if !isempty(args) && (arg = first(args); !isempty(arg) && arg[1] == '-' && in(arg, reqarg))
-            println(STDERR, "julia: option `$arg` is missing an argument")
-            exit(1)
+        if !isempty(args)
+            arg = first(args)
+            if !isempty(arg) && arg[1] == '-' && in(arg, reqarg)
+                println(STDERR, "julia: option `$arg` is missing an argument")
+                exit(1)
+            end
+            idxs = find(x -> x == "--", args)
```

Comment:
> gives an example

Do you mean to use `--` to allow option like file name? Or allowing nesting it.

For the first one, I think most coreutils have that.

For the second one, at least the ArchLinux dev-tools is using it to allow passing arguments to different components in the chroot toolchain.